### PR TITLE
calculate risk to gentrification and flag for different stages

### PR DESCRIPTION
**Pull request by maggiesu0725**

### CODE REVIEW

**Comment by maggiesu0725** [comment]
@rpitingolo could you take a look at the average income variable? the change seems unbelievably large for some tracts

**Comment by rpitingolo** [comment]
@maggiesu0725 which dataset in the vulnerable to gentriciation.sas program should I be looking at?

**Comment by maggiesu0725** [comment]
@rpitingolo merged and changeintime, specifically, when I run the proc means on changeintime, I get hte averagechange of income for washington region during 2000-2016 is 41637, which is high (maybe not?)

**Comment by maggiesu0725** [comment]
@rpitingolo I didn't adjust for inflation.

**Comment by maggiesu0725** [comment]
@lhendey see %dollar_convert part

**Comment by maggiesu0725** [comment]
@lhendey @rpitingolo the program for Lisa Bates typology of gentrification analysis. Might need changes, but main part ready for review

**Review by lhendey — CHANGES_REQUESTED** [review]
@maggiesu0725 - please see comments here and in the word doc i emailed. cc @rpitingolo

File: Prog/Vulnerable to gentrification.sas [hunk 1]
```diff
@@ -24,118 +23,223 @@
...
- 
+data merged;
+merge demographics16 demographics00;
+by geo2010;
+COG=1;
+run;
+
+data changeintime;
+set merged;
+deltarenter = numrenteroccupiedhu_&_years.- sprntoc0;
```
[the remark this answers is not in the document]
Comment (maggiesu0725) [reply Prog/Vulnerable to gentrification.sas]: yes

File: Prog/Vulnerable to gentrification.sas [hunk 2]
```diff
@@ -24,118 +23,223 @@
...
+percentrenter_2016= numrenteroccupiedhu_&_years./(numrenteroccupiedhu_&_years.+numowneroccupiedhu_&_years.);
+percentwhite_2016= popwhitenonhispbridge_&_years./popwithrace_&_years.;
+percentcollege_2016= pop25andoverwcollege_&_years./pop25andoveryears_&_years.;
+avghhinc_2016= agghshldincome_&_years./(numhshlds_&_years. );
 run;
 
-%let Ratio_rentgrs_rent_12_16 = 1.1512114;         %** Value copied from Proc Means output **;
+/* identify tracts with higher than average population with characteristics that make resisting displacement more difficult:
+renters, POC, lack college degree, lower income*/
+/*median for 2016: percentwhite:0.44 percentrenter: 0.38 percentcollege 0.51 averageincome 123451*/
```
[the remark this answers is not in the document]
Comment (maggiesu0725) [reply Prog/Vulnerable to gentrification.sas]: Do you mean 0.4 instead of 0.44?

[the remark this answers is not in the document]
Comment (lhendey) [reply Prog/Vulnerable to gentrification.sas]: no - 0.443 instead of .44

**Comment by maggiesu0725** [comment]
could you take a look at Vulnerable to gentrification_Jurisdiction? I wrote a macro but it's not running, don't know where to debug. @lhendey

**Comment by lhendey** [comment]
@maggiesu0725 - is this ready for review?

**Comment by maggiesu0725** [comment]
It generates the results we want, but I haven’t managed to consolidate the program to make it concise and pretty yet. @lhendey

**Comment by lhendey** [comment]
@maggiesu0725 please review the changes i made in https://github.com/NeighborhoodInfoDC/RegHsg/pull/9/commits/40eb1df61227d3e37f54aa51379fdac2bf3417b0 

some were for programming efficiency and some were correcting errors.  

i'd like to also have a version that uses median HH income (not median family income) and see what the sensitivity is. if we don't have that in the currect ACS data please go through the process to add it. for 2000 please look it up for the jurisdictions in american fact finder SF3 table HCT012.  

the adjacency process will need to be repeated. i didn't make all the changes in the regionwide program (which we are no longer using, right?)

**Comment by maggiesu0725** [comment]
Thanks, I'll try. @lhendey

**Comment by maggiesu0725** [comment]
@lhendey alternative calculation by median hh income is updated. I calculated and there seems no difference between the two in terms of the final flag.

**Comment by lhendey** [comment]
@maggiesu0725  - i made some changes. please run the adjacency in ARCGIS to complete the typology.

**Comment by lhendey** [comment]
@maggiesu0725  - there are two demographic change flags now.

**Comment by maggiesu0725** [comment]
@lhendey I added the HH count and HH under 80% AMI.

**Comment by lhendey** [comment]
@maggiesu0725 -  

I decided to use the 2013-17 data so converted all date references to macro vars in code. 

your adjacent flag file is bringing in some tracts outside the COG region please clip before. 

i've added a flag "potentialADJ" - 
if rank2017 ~=. and rank00_17 ~=. then do;
	if  rank2017<=2 & rank00_17 <=2 then potentialADJ=1; else potentialADJ=0;
	end;
else if rank2017=. or rank00_17=. then potentialADJ=.n;  

correct? 

You should use that to then determine adjacency to high 2017 or high2000-2017 tracts. I also need to be able to see that in ARC GIS. I'm not sure from what i can see on L? (to do some QC on the adjacency). '

i added code to calculate the typology for either median family income or median household income. 

i also added some code at the end to output files we could use in tables. 

I think we are really close. If you can re-run the adjacency - using the 2013-17 data and the potentialADJ flag, I can take a look at the ARC GIS file and then we can merge the updated adjacency back in and finalize this. 

Thanks!

**Comment by lhendey** [comment]
@maggiesu0725 - i'm getting errors trying to run the R file.

**Pull request merged**